packages/flutter_tools/lib/src/isolated/native_assets/native_assets.dart

```diff
@@ -435,6 +340,28 @@ Future<Uri> _writeNativeAssetsYaml(
   return nativeAssetsFile.uri;
 }
 
+String _toNativeAssetsJsonFile(List<KernelAsset> kernelAssets) {
+  final Map<Target, List<KernelAsset>> assetsPerTarget = <Target, List<KernelAsset>>{};
+  for (final KernelAsset asset in kernelAssets) {
+    final List<KernelAsset> assets = assetsPerTarget[asset.target] ?? <KernelAsset>[];
+    assets.add(asset);
+    assetsPerTarget[asset.target] = assets;
+  }
+
+  // See assets/native_assets.cc in the engine for the expected format.
+  final Map<String, Object> jsonContents = <String, Object>{
+    'format-version': <int>[1, 0, 0],
```

Comment:
This version should be a constant somewhere and so should the keys. Also, could this not just be a string?

Reply:
String literals are already constants.

I'm not sure the code would become more readable with the keys outlined.

> Also, could this not just be a string?

The version? It's a list of numbers, not a string. The list of numbers avoids needing string operations to check.